### ISSUE TITLE
Not reading with macOS

### ISSUE THREAD

**Issue by hippee-lee** [jence/j4210u-app#16]
Hello,

I'm not able to connect with the macOS app but when I scan, nothing is read and there is no indication from the reader that it is scanning. (green light does not light up).

I'm happy to try and look into it, not a java expert but I know my way around code if you are able to give me some pointers to navigating the codebase.

**Comment by jence** [comment]
One of the most common reason for this to happen in Mac OS X is if you are using Intel Silicon. We didn't have any customer complain about having any issue with Apple Silicon.

We are getting increasing difficulty in supporting Intel Silicon because Apple no long support the older version of Intel based Mac OS X and Apps developed on M1 Mac does not work on Intel Mac. We have tried to make a special build for individual customers so the App works on Intel Silicon.

Please provide the following so we can make a special build for your Intel Mac. Then you can try it.

1. Your Mac's OS Version: You can get this information from Mac's Apple Menu.

2. Your Java version: Open a Terminal windows and type (send us a screenshot, the text that is displayed) :

    java -version

Once you provide this information, we will build a j4210u.dylib for you and a full Mac distribution for you to try on Intel.

**Comment by hippee-lee** [comment]
I'm running on Apple silicon. 

OS: `13.4 (22F66)`
Chip: `Apple M1 Max`
Java: 
```
java --version
openjdk 17.0.7 2023-04-18
OpenJDK Runtime Environment Temurin-17.0.7+7 (build 17.0.7+7)
OpenJDK 64-Bit Server VM Temurin-17.0.7+7 (build 17.0.7+7, mixed mode)
```

I've tried both the prepackaged app and a local build from source.

**Comment by jence** [comment]
Thank you for the info.
I have tested in the same configuration as yours and now it's working. Please get the latest from the main branch. Please let me know if this fixes the problem. If Mac OS X block your problem by displaying this window:
![Screenshot 2023-06-28 at 8 32 16 PM](https://github.com/jence/j4210u-app/assets/5038489/f2bcd1a4-003f-416a-8718-54c965deb46e)
Go to Privacy & Security option from the System Settings and a click on Open Anyway.
Mac OS X based on Intel chip is now in a separate folder macosx[intel].
Regards,
Mansur.

**Comment by hippee-lee** [comment]
Thanks for separating the two apps. When running the app compiled for apple silicon I see the device connected via usb port when I refresh. But when I scan I don't hear anything or get any readings off the RFID tag on my car or the sample tag provided. 

I do hear a beep noise and the led light flashes orange/amber when I first connect the device/hardware to the computer - I assume this is a power on signal?

**Comment by jence** [comment]
After connecting by clicking on Connect button, please click on the Info tab. If you are able to view the Info tab, then the connection is okay. If an error message pops up, then there is a serial communication error. If you consistently get this error message, then check if this happen in Windows PC as well (of course, if you have a Windows PC available). Thanks.

**Comment by hippee-lee** [comment]
It looks like the connection is fine. Attached is a screen capture of the into tab.
<img width="1410" alt="Screenshot 2023-07-06 at 12 17 10 PM" src="https://github.com/jence/j4210u-app/assets/433692/3d354bce-b8fc-460f-857f-5bcb78d70841">

As soon as I click the scan button the application locks up and I have to force quit it.

**Comment by jence** [comment]
Hi Hippee,
Please download the directory from this link. This is what I am using. So, you and I will on same page. This works for me. Please check if this works on your Mac. 
https://drive.google.com/file/d/1WdwU13RJ9NDdvRf9PLa0keyfjWeK7s6T/view?usp=share_link
Regards,
Mansur.

**Comment by hippee-lee** [comment]
The app from the download connects and doesn't lockup when I try to scan. But it's not reading anything when I try to scan the tag that was provided with the unit I purchased. Should that be readable?

**Comment by jence** [comment]
Yes, it should read the tag that was provided. The zip I sent works for me.
Now you can try the following:

1. Do not use a hub, instead attach to USB directly.
2. Try different USB ports and check if the result is the same.
3. Try sudo UhfApp.app from a terminal window, i.e., try to run the program
as root.
4. If you have another UHF tag, you can try that too. Just to make sure
that the device works, but the tag was bad. If you do not have any more
tag, you can order from here:
Dogbone Long Range Tag: https://www.amazon.com/dp/B0C6YRZ8VT
Anti Theft Jewlery Tag: https://www.amazon.com/dp/B0C7YBC2CW
These are sold by us with 30 day Amazon return policy.

I am trying to figure out what is causing the issue. So, be patient, we
will figure out the issue.
Thanks.

Regards,
Mansur.

**Comment by hippee-lee** [comment]
I will try your suggestions as soon as I can, might have to be this weekend though. Than you for your continued support, I am grateful for it.

**Comment by jence** [comment]
Please try the latest version, which is 2.2.1. This fixes a bug in the libj4210u.dylib library. It is committed to GitHub. Thank you.
Mansur.

**Comment by hippee-lee** [comment]
Hi Mansur,

I tried v2.2.1 but same results. There is a beep from the unit when I plug it into the usb port and it connects. But none of the tags I have nor the one that came with the unit are scanning. The app just says "No tags found". 

I'm debating if I should return the unit or not. I ordered the dogbone tags to test with other tags before I do that. I'd really like to get this working.

**Comment by jence** [comment]
You can always return or exchange. I tried to fix it. A replacement will
give us a chance to figure out what the real problem was. We are interested
to find the issue causing it. Does not matter if you return it.

Thank you for trying our product.
-Mansur.

**Comment by hippee-lee** [comment]
Ok, if I want to exchange it should I do something outside of Amazon? Or is it just easier to return to Amazon and then purchase a new one? 

I'm very grateful for all of your help trying to get it to work on **_my_** macOS w/ Apple silicon. 🙏🏼

**Comment by jence** [comment]
Returning to Amazon and then buying again is much better. When you return,
mark it as defective. Amazon sends the defective devices back to us. We
test each device and learn about the issue the customer was facing. Each
returned device give us learning opportunity. Our software became much
stable now. I am eager to check your device because we cannot reproduce the
problem you are facing.

So please return the device as defective, then buy again. Thank you for
your support. Not every one are as patient as you are.

-Mansur.

**Comment by hippee-lee** [comment]
Sounds good. I ordered some of the rfid tags you suggested and will test with them. If it doesn't work I'll return as suggested. 

I'm a software engineer so I understand software and hardware development. Thanks for supporting your product.
